### PR TITLE
Loockup msg server test

### PR DESCRIPTION
**Pull request by hieuvubk**
Closes: #2183

## What is the purpose of the change

> Add tests for lockup module msg_server.test. Add some missing ValidateBasic.

- [x] MsgBeginUnlockingAll

- [x] MsgBeginUnlocking

- [x] MsgExtendLockup

### CODE REVIEW

File: x/lockup/types/msgs.go
Comment (czarcas7ic) [inline x/lockup/types/msgs.go:114]:
```suggestion
	// only allow unlocks with a single denom
```

File: x/lockup/types/msgs_test.go
Comment (czarcas7ic) [inline x/lockup/types/msgs_test.go:102]: Seems like we need to generate these three vars for every test. Maybe we can create a function to do this to reduce boilerplate?

File: x/lockup/types/msgs_test.go
Comment (czarcas7ic) [inline x/lockup/types/msgs_test.go:142]:
```suggestion
```

expectPass defaults to false, so every occurrence of false can be removed from all the tests to make the tests shorter.

**Review by czarcas7ic — APPROVED** [review]
LGTM! 👍

File: x/lockup/types/msgs_test.go
Comment (hieuvubk) [reply x/lockup/types/msgs_test.go]: Yes thanks for the suggestion. I created a function to gen valid & invalid addresses for all tests

**Comment by czarcas7ic** [comment]
This all LGTM, much cleaner! Going to wait on one more review in case I missed anything then should be good to merge!

File: x/lockup/types/msgs_test.go
Comment (p0mvn) [inline x/lockup/types/msgs_test.go:113]: Can we please instead make the assertions inside the for loop for each test case?

Also, let's try to avoid the complexity with `createMsg` callbacks, please. I think we can directly initialize:
```
msg: MsgBeginUnlockingAll{
			Owner: addr1,
}
```
instead of having the callback overhead

File: x/lockup/types/msgs_test.go
Comment (p0mvn) [inline x/lockup/types/msgs_test.go:138]: Let's make sure each test case has its own context pelase

`t.Run(test.name, ...)`

**Review by p0mvn — CHANGES_REQUESTED** [review]

File: x/lockup/types/msgs_test.go
Comment (hieuvubk) [reply x/lockup/types/msgs_test.go]: Yes thank you. I think it makes the code look better.

File: x/lockup/types/msgs_test.go
Comment (p0mvn) [inline x/lockup/types/msgs_test.go:254]: Can we move this helper to test utils?

File: x/lockup/types/msgs_test.go
Comment (p0mvn) [inline x/lockup/types/msgs_test.go:194]: Does it make sense to define these tests on the test suite?

**Review by p0mvn — APPROVED** [review]
LGTM, great work @hieuvubk 

I have 2 follow-up questions. If questions make sense, can you please either address them or create issues for them to be addressed? Thank you

File: x/lockup/types/msgs_test.go
Comment (hieuvubk) [reply x/lockup/types/msgs_test.go:254]: Yes I'll do it now

File: x/lockup/types/msgs_test.go
Comment (hieuvubk) [reply x/lockup/types/msgs_test.go:194]: Do u mean to use the test suite instead of testing.T?

Comment (p0mvn) [reply x/lockup/types/msgs_test.go:193]: Sorry for not being clear enough.

What I meant is to define something like this:
https://github.com/osmosis-labs/osmosis/blob/fe98f6e4453cd32035d277c21ef2f3669b677bb2/x/lockup/keeper/keeper_test.go#L15

Instead, we would have `TypesTestSuite`

File: x/lockup/types/msgs_test.go
Comment (p0mvn) [reply x/lockup/types/msgs_test.go]: What I meant is to define it on the app testing suite:
https://github.com/osmosis-labs/osmosis/blob/450e837158f4ae4cb908ec496da47fc169007218/app/apptesting/test_suite.go#L273

If you go with my other suggestion of [utilizing test suite](https://github.com/osmosis-labs/osmosis/pull/2205#discussion_r933548244), it would be convenient to reuse

File: x/lockup/types/msgs_test.go
Comment (hieuvubk) [reply x/lockup/types/msgs_test.go:193]: I think we should create an issue for that. I just took a look through all modules and tests in `types` all use testing.T
Should refactor all those tests in a pr right?

Comment (p0mvn) [reply x/lockup/types/msgs_test.go:192]: If that's the case, let's keep this as is. I think consistency is more important here. Thanks for double-checking that

**Review by p0mvn — APPROVED** [review]
:rocket:

**Comment by p0mvn** [comment]
Merging since @czarcas7ic previously approved this and only my comments have been applied since

**Pull request merged**
